symengine/cwrapper.cpp

```diff
@@ -200,6 +205,34 @@ double real_double_get_d(const basic s)
     return (rcp_static_cast<const RealDouble>(s->m))->as_double();
 }
 
+#ifdef HAVE_SYMENGINE_MPFR
+
+void real_mpfr_set_d(basic s, double d, int prec)
+{
+    mpfr_t mp;
```

Comment:
You can access the `mpfr_t` directly. This way, `mp` is not copied unnecessarily.